pkg/util/kubernetes/apiserver/util.go

```diff
@@ -61,3 +67,35 @@ func UnstructuredFromWPA(structIn *v1alpha1.WatermarkPodAutoscaler, unstructOut
 	unstructOut.SetUnstructuredContent(content)
 	return nil
 }
+
+func retryHandler(informers map[InformerName]cache.SharedInformer, name InformerName, config retryConfig) error {
+	nextTry := config.InitialRetryDelay
+	lastTry := false
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), nextTry)
+		start := time.Now()
+		if !cache.WaitForCacheSync(ctx.Done(), informers[name].HasSynced) {
+			log.Warnf("couldn't sync informer %s in %v", name, time.Now().Sub(start))
+			nextTry = config.method(nextTry, config)
+			config.RetryCount++
+			log.Warnf("Informers failed to sync, increasing timeout to %s", nextTry)
+			if lastTry {
+				cancel()
+				return fmt.Errorf("couldn't sync informer %s in %v", name, time.Now().Sub(start))
+			}
+			if nextTry >= config.MaxRetryDelay {
+				nextTry = config.MaxRetryDelay
+				lastTry = true
+			}
+		} else {
+			cancel()
```

Comment:
The only think worrying me with the non deferred cancel is in case of panics we don't release the resources, but I do think that the current approach is correct 👍